### PR TITLE
[Enhancement] add cache metrics

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

This pr add restful api for query cache management and monitoring.
1. monitoring: 
```
curl -s  http://${be_host}:${be_http_port}/metrics |grep query_cache

# TYPE starrocks_be_query_cache_capacity gauge
starrocks_be_query_cache_capacity 536870912
# TYPE starrocks_be_query_cache_hit_count gauge
starrocks_be_query_cache_hit_count 5084393
# TYPE starrocks_be_query_cache_hit_ratio gauge
starrocks_be_query_cache_hit_ratio 0.984098
# TYPE starrocks_be_query_cache_lookup_count gauge
starrocks_be_query_cache_lookup_count 5166553
# TYPE starrocks_be_query_cache_usage gauge
starrocks_be_query_cache_usage 0
# TYPE starrocks_be_query_cache_usage_ratio gauge
starrocks_be_query_cache_usage_ratio 0.000000
```
2. query cache management
```
## show stat of query cache
curl  http://${be_host}:${be_http_port}/api/query_cache/stat
{
    "capacity": 536870912,
    "usage": 0,
    "usage_ratio": 0.0,
    "lookup_count": 5025124,
    "hit_count": 4943720,
    "hit_ratio": 0.983800598751394
}

## clear cache by invaliating all cache entries.
curl  -XPUT http://${be_host}:${be_http_port}/api/query_cache/invalidate_all

{
    "status": "OK"
}

```

### CODE REVIEW

File: be/src/exec/query_cache/cache_manager.cpp [hunk 1]
```diff
@@ -41,5 +41,19 @@ size_t CacheManager::capacity() {
...
+}
+
+size_t CacheManager::hit_count() {
+    return _cache.get_hit_count();
+}
+
+void CacheManager::clear() {
+    auto old_capacity = _cache.get_capacity();
+    _cache.set_capacity(0);
+    _cache.set_capacity(old_capacity);
```
Comment (kangkaisen) [inline be/src/exec/query_cache/cache_manager.cpp:55]: ?

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: be/src/http/action/query_cache_action.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+namespace starrocks {
+
+const static std::string HEADER_JSON = "application/json";
+const static std::string ACTION_KEY = "action";
+const static std::string ACTION_STAT = "stat";
+const static std::string ACTION_INVALIDATE_ALL = "invalidate_all";
+
+void QueryCacheAction::handle(HttpRequest* req) {
+    LOG(INFO) << req->debug_string();
```
Comment (silverbullet233) [inline be/src/http/action/query_cache_action.cpp:26]: should use debug level log?

File: be/src/exec/query_cache/cache_manager.cpp [hunk 3]
```diff
@@ -41,5 +41,20 @@ size_t CacheManager::capacity() {
...
+}
+
+size_t CacheManager::hit_count() {
+    return _cache.get_hit_count();
+}
+
+void CacheManager::invalidate_all() {
+    auto old_capacity = _cache.get_capacity();
+    // set capacity of cache to zero, the cache shall prune all cache entries.
+    _cache.set_capacity(0);
```
Comment (kangkaisen) [inline be/src/exec/query_cache/cache_manager.cpp:55]: why not call prune method, then we needn't the comment.

**Review by kangkaisen — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**